schemas/constructs/component/v1beta1/component.json

```diff
@@ -0,0 +1,115 @@
+    },
+    "metadata": {
+      "type": "object",
+      "description": "Metadata contains additional information associated with the component.",
+      "required": ["name", "version"],
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "Various properties and characteristics of the component."
+        },
+        "capabilities": {
+          "type": "object",
+          "description": "Meshery manages components in accordance with their specific capabilities. This field explicitly identifies those capabilities largely by what actions a given component supports; e.g. metric-scrape, sub-interface, and so on. This field is extensible. ComponentDefinitions may define a broad array of capabilities, which are in-turn dynamically interpretted by Meshery for full lifecycle management."
+        },
+        "genealogy": {
+          "type": "string",
+          "description": "Genealogy represents the various representational states of the component."
+        },
+        "isAnnotation": {
+          "type": "boolean",
+          "description": "Identifies whether the component should be treated as deployable entity or just for logical representation."
+        },
+        "shape": {
+          "type": "string",
+          "description": "Shape of the component used for UI representation.",
+          "enum": ["ellipse","triangle","round-triangle","rectangle","round-rectangle","bottom-round-rectangle","cut-rectangle","barrel","rhomboid","diamond","round-diamond","pentagon","round-pentagon","hexagon","round-hexagon","concave-hexagon","heptagon","round-heptagon","octagon","round-octagon","star","tag","round-tag","vee"]
+        },
+        "primaryColor": {
+          "type": "string",
+          "description": "Primary color of the component used for UI representation."
+        },
+        "secondaryColor": {
+          "type": "string",
+          "description": "Secondary color of the component used for UI representation."
+        },
+        "svgWhite": {
+          "type": "string",
+          "description": "White SVG of the component used for UI representation on dark background."
+        },
+        "svgColor": {
+          "type": "string",
+          "description": "Colored SVG of the component used for UI representation on light background."
+        },
+        "svgComplete": {
+          "type": "string",
+          "description": "Complete SVG of the component used for UI representation, often inclusive of background."
+        },
+        "published": {
+          "type": "boolean",
+          "description": "Controls whether the component should be registered in Meshery Registry."
+        }
+      },
+      "additionalProperties": true
+    },
+    "entity": {
```

Comment:
Since we are inside of the component.schema, I propose that we call this "component", not "entity". We *will* continue to refer to all model consituent objects as "entities" as a generic term, though.